### DISCUSSION TITLE
How to bind  buffer slices in a loop?

### DISCUSSION THREAD

**Discussion by adamnemecek** [gfx-rs/wgpu#1433] [General]
Hey gang,
I'm not sure how to go about the following code.
Let's say, I have some struct `Layer` which contains a vertex buffer of triangles. I have a `Vec` of these Layers and want to render them in a loop like so:
```rust
for layer in layers {
    pass.set_vertex_buffer(0, layer.vertex_buffer().as_slice(..));
    pass.draw(0..3, layer.len());
}
```

The lifetime requirements make it non-obvious how to use this since when you call `as_slice` you get a temporary.

**Comment by kvark** [comment]
> since when you call as_slice you get a temporary.

This doesn't sound right. A slice is never a temporary - it has the lifetime of the host, and it's borrowing the host.

What is the type returned by `vertex_buffer`?
Also, what is the compile error you are getting?
